diff --git a/kernel/sched/extension/eas_plus.c b/kernel/sched/extension/eas_plus.c
index 8178a66cb..71526eb99
--- a/kernel/sched/extension/eas_plus.c
+++ b/kernel/sched/extension/eas_plus.c
@@ -209,7 +209,7 @@ inline unsigned int cpu_is_fastest(int cpu)
 	struct list_head *pos;
 
 	if (!pod_is_ready()) {
-		printk_deferred("Perf order domain is not ready!\n");
+		printk_deferred_once("Perf order domain is not ready!\n");
 		return -1;
 	}
 
@@ -224,7 +224,7 @@ inline unsigned int cpu_is_slowest(int cpu)
 	struct list_head *pos;
 
 	if (!pod_is_ready()) {
-		printk_deferred("Perf order domain is not ready!\n");
+		printk_deferred_once("Perf order domain is not ready!\n");
 		return -1;
 	}
 
@@ -238,7 +238,7 @@ bool is_intra_domain(int prev, int target)
 	struct perf_order_domain *perf_domain = NULL;
 
 	if (!pod_is_ready()) {
-		printk_deferred("Perf order domain is not ready!\n");
+		printk_deferred_once("Perf order domain is not ready!\n");
 		return 0;
 	}
 
